src/node/node_state.h

```diff
@@ -1347,7 +1355,9 @@ namespace ccf
               "Unexpected: write to service table does not include key 0");
           }
 
-          if (it->second.value().status == ServiceStatus::OPEN)
+          if (
+            it->second.value().status == ServiceStatus::OPEN &&
+            this->is_part_of_network())
```

Comment:
On recovery, we used to open the user frontend way too early when we were replaying the previous service opening.